theia-extensions/viewer-prototype/src/node/trace-server-service.ts

```diff
@@ -1,57 +1,65 @@
-import { spawn } from 'child_process';
+import { spawn, ChildProcess } from 'child_process';
 import { injectable } from 'inversify';
 import { PortBusy, TraceServerConfigService } from '../common/trace-server-config';
 import treeKill = require('tree-kill');
+
 @injectable()
 export class TraceServerServiceImpl implements TraceServerConfigService {
-    private processId: number;
 
-    startTraceServer(path: string | undefined, port: number | undefined): Promise<string> {
-        const server = spawn(`${path}`, ['-vmargs', `-Dtraceserver.port=${port}`]);
-        this.processId = server.pid;
-        const timeouts: NodeJS.Timeout[] = [];
+    private server?: ChildProcess;
 
+    async startTraceServer(path?: string, port?: number): Promise<string> {
+        if (path === undefined) {
+            throw new Error('no trace server path specified');
+        } else if (this.server && this.isServerRunning(this.server)) {
+            throw new Error('trace server is already running');
+        }
+        // TODO: Since the resolve never actually gets returned, we need a better way to determine if the child process
+        // initiated by the spawn successfully starts the trace server.
+        const server = spawn(path, ['-vmargs', `-Dtraceserver.port=${port}`]);
+        if (server.pid === undefined) {
+            // When pid is undefined this means we're about to get an error:
+            return new Promise<never>((resolve, reject) => server.once('error', reject));
+        }
+        this.server = server;
+        const timeouts: NodeJS.Timeout[] = [];
```

Comment:
Is this change required for the upgrade or is it an improvement? I think it should be outside of the scope of this PR of upgrading Theia then. Could you have that as a separate PR after the upgrade?

Reply:
Upgrading TypeScript caused an issue with some of the previous code, and that caused me to drive-by this code as well... I'll provide a dirty fix instead that just tells TypeScript to ignore the previous errors for now I guess.